### PR TITLE
trait - simplify method with null coalescing operator

### PR DESCRIPTION
**Pull request by k-timoshenko**

### CODE REVIEW

**Review by MatthewPattell — COMMENTED** [review]
Add please return type for all methods :)

File: ImplementServices.php [hunk 1]
```diff
@@ -77,31 +77,30 @@ public function __call($name, $params)
      * @param string $name
      * @param array  $params
      *
-     * @return null
+     * @return BaseService|null
      */
-    private function getService(string $name, array $params = [])
+    private function getService(string $name, array $params = []): ?BaseService
     {
         $className = $this->getClassName($name);
```
Comment (MatthewPattell) [inline ImplementServices.php]: add a line break

File: ImplementServices.php [hunk 2]
```diff
@@ -77,31 +77,30 @@ public function __call($name, $params)
...
+        }
 
-        if ($className) {
-            if (!isset($this->servicesInstances[$name])) {
-                $this->servicesInstances[$name] = new $className($this, isset($params[0]) && is_array($params[0]) ? $params[0] : []);
-            }
-
-            return $this->servicesInstances[$name];
+        if (!isset($this->servicesInstances[$name])) {
+            $this->servicesInstances[$name] = new $className($this, isset($params[0]) && is_array($params[0]) ? $params[0] : []);
```
Comment (MatthewPattell) [inline ImplementServices.php]: Can shorten this line: is_array($params[0] ?? '')
Thx

File: ImplementServices.php (hunk 1 above)
Comment (MatthewPattell) [inline ImplementServices.php:84]: add a line break ^)

Comment (MatthewPattell) [reply ImplementServices.php:84]: And change php version in composer: 7.1 (Nullable RFC)

**Comment by k-timoshenko** [comment]
Done.
php now min 7.1
yii 2.0.12 because of BaseObject

**Pull request merged**